### PR TITLE
OS-8353 piadm should get the download location from a config file

### PR DESCRIPTION
**Pull request by danmcd**
WORK IN PROGRESS.

### CODE REVIEW

**Comment by danmcd** [comment]
Testing notes in the ticket.

File: src/piadm.sh [hunk 1]
```diff
@@ -176,11 +176,68 @@ vcurl() {
...
+		mkdir -p /var/piadm
+
+		# The latest default behavior should be hardcoded in this path.
+		# Update as we add more things.  Each change bumps
+		# PIADM_CONFIG_VERSION by one.
+		PIADM_CONFIG_VERSION=1
+		printf "PIADM_CONFIG_VERSION=%s\n" $PIADM_CONFIG_VERSION > \
+			$PIADM_CONF
+		printf "DEFAULT_URL_PREFIX=%s\n" $DEFAULT_URL_PREFIX >> \
+			$PIADM_CONF
```
Comment (bahamat) [inline src/piadm.sh:213]: Consider a heredoc instead of multiple redirects.

File: src/piadm.sh [hunk 2]
```diff
@@ -176,11 +176,68 @@ vcurl() {
...
+config_check() {
+	# Can't do standalone_only per se as it errors out, but this only
+	# applies to standalone SmartOS.
+	if [[ "$TRITON_CN" == "yes" || "$TRITON_HN" == "yes" ]]; then
+		return
+	fi
+
+	OLD_DEFAULT="$DEFAULT_URL_PREFIX"
+	if [[ -f $PIADM_CONF ]]; then
+		. $PIADM_CONF
```
Comment (bahamat) [inline src/piadm.sh:197]: I prefer `source <file>` over `. <file>` because it's `grep`able.

**Review by bahamat — COMMENTED** [review]
Does this still let you override with an environment variable?

**Comment by danmcd** [comment]
> Does this still let you override with an environment variable?

Yes, see line 239.

File: src/piadm.sh (hunk 2 above)
Comment (danmcd) [reply src/piadm.sh:197]: I thought `source` was csh/tcsh/zsh only.

File: src/piadm.sh (hunk 1 above)
Comment (danmcd) [reply src/piadm.sh:213]: Like:
```
cat <<<FILE
PIADM_CONFIG_VERSION=$PIADM_CONFIG_VERSION
...
FILE > $PIADM_CONF
```
?

Comment (bahamat) [reply src/piadm.sh:213]: Just `<<` not `<<<`, but otherwise yeah.

```
cat << FILE
stuff
FILE > $PIADM_CONF
```

File: src/piadm.sh (hunk 2 above)
Comment (bahamat) [reply src/piadm.sh:197]: It works in bash too.

Comment (danmcd) [reply src/piadm.sh:197]: This file already uses `. <PATH>` elsewhere, not changing this one.

File: src/piadm.sh (hunk 1 above)
Comment (danmcd) [reply src/piadm.sh:213]: Will fix.

**Review by bahamat — APPROVED** [review]

**Pull request merged**